src/storage-client/src/source/persist_source.rs

```diff
@@ -143,9 +154,51 @@ where
         Arc::new(metadata.relation_desc),
         Arc::new(UnitSchema),
         move |stats| {
-            mfp_pushdown.as_ref().map_or(true, |x| {
-                x.should_fetch(&PersistSourceDataStatsImpl { desc: &desc, stats })
-            })
+            if let Some(plan) = &filter_plan {
+                let arena = RowArena::new();
+                let mut ranges = ColumnSpecs::new(desc.typ().clone(), &arena);
```

Comment:
I think this all needs to be fake_desc instead of desc

Reply:
Ugh, this was intentional but I can no longer remember why! If I can't figure it out I'll switch it back.

Reply:
On reflection I think the rationale is that by the time we're calling `MFP::evaluate` on a particular `Row`, the type of the row is described by the actual `RelationType` and not the hack. So I think this is the correct thing!